modules/assim.sequential/R/sda.enkf.R

```diff
@@ -129,176 +145,382 @@ sda.enkf <- function(settings,IC,prior,obs){
       
   ## start model run
   start.model.runs(settings,settings$database$bety$write)
-  
-
-  time = start.year:end.year
-  nt = length(time)
-  ens = list()
-  NPPm = rep(NA,nens)
+   
+  total.time = as.numeric(end.year):(as.numeric(end.year)+nrow(obs.mean))
```

Comment:
missing the logic here about how the number of observations determines the total.time, and why the counter starts from end.year not start.year. Is this an attempt to generalize to the case where observations are at a different frequency than once per year?